### PR TITLE
Credential Builder

### PR DESCRIPTION
**Pull request by decentralgabe**
#31

### CODE REVIEW

**Comment by decentralgabe** [comment]
May add some more tests before merge. I'm happy that the tests pass against the spec's example.

**Comment by decentralgabe** [comment]
Added tests for the builder

File: cryptosuite/jwssignaturesuite.go [hunk 1]
```diff
@@ -56,40 +61,37 @@ func (j JWSSignatureSuite) RequiredContexts() []string {
...
-
+func (j JWSSignatureSuite) Sign(s Signer, p Provable) error {
 	// create proof before CVH
 	proof := j.createProof(s.KeyID())
 
 	// prepare proof options
 	contexts, err := GetContextsFromProvable(p)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get contexts from provable")
+		return errors.Wrap(err, "could not get contexts from provable")
```
Comment (mistermoe) [inline cryptosuite/jwssignaturesuite.go:71]: 🎉 meaningful error messages

File: did/model_test.go [hunk 2]
```diff
@@ -3,9 +3,10 @@
 package did
 
 import (
-	"encoding/json"
 	"testing"
 
+	"github.com/goccy/go-json"
```
Comment (mistermoe) [inline did/model_test.go:8]: solid choice! being able to pass Context.context into Marshal and Unmarshal is awesome

File: vc/credential.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
+package vc
```
Comment (mistermoe) [inline vc/credential.go:1]: I think it'd be helpful to include package level documentation ([reference](https://talks.golang.org/2013/bestpractices.slide#15)). Can also use the `doc.go` convention

Comment (mistermoe) [inline vc/credential.go:1]: would it make more sense if this file was called `builder.go` given that's what's in it?

**Review by mistermoe — APPROVED** [review]
Overall, looks great! left a few nits.

File: cryptosuite/jsonwebkey2020.go [hunk 4]
```diff
@@ -414,7 +417,7 @@ func (v JSONWebKeyVerifier) Verify(message, signature []byte) error {
 	return err
 }
 
-func NewJSONWebKeyVerifier(key PublicKeyJWK) (*JSONWebKeyVerifier, error) {
+func NewJSONWebKeyVerifier(kid string, key PublicKeyJWK) (*JSONWebKeyVerifier, error) {
```
[the remark this answers is not in the document]
Comment (decentralgabe) [reply cryptosuite/jsonwebkey2020.go:420]: for linked data things, there's an expected `verificationMethod` that indicates a key-reference in a DID, or another URI-resolvable document; this is only required for the signer. The verifier _technically_ does not need this; however, the JWK library we use requires the `kid` in the public key you construct to match the `kid` in the signed message you're verifying...which is why on this specific verifier it's a required parameter.

File: vc/credential.go (hunk 3 above)
Comment (decentralgabe) [reply vc/credential.go:1]: opened https://github.com/TBD54566975/did-sdk/issues/40

Comment (decentralgabe) [reply vc/credential.go:1]: will rename

**Pull request merged**